topydo/ui/TodoListWidget.py

```diff
@@ -101,132 +119,199 @@ def _scroll_to_bottom(self, p_size):
         self.listbox.calculate_visible(p_size)
 
     def keypress(self, p_size, p_key):
-        # first check whether 'g' was pressed previously
-        if self.keystate == 'g':
-            if p_key == 'g':
-                self._scroll_to_top(p_size)
+        urwid.emit_signal(self, 'remove_pending_action')
+        requires_further_input = ['postpone', 'postpone_s', 'pri']
 
-            # make sure to accept normal shortcuts again
-            self.keystate = None
-            return
-        elif self.keystate in ['p', 'ps']:
-            if p_key not in ['d', 'w', 'm', 'y']:
-                if p_key.isdigit():
-                    self._pp_offset += p_key
-                elif self.keystate == 'p' and p_key == 's':
-                    self.keystate = 'ps'
-                else:
-                    self._pp_offset = ''
-                    self.keystate = None
+        keymap, keystates = self.keymap
+
+        shortcut = self.keystate or ''
+        if len(p_key) > 1:
```

Comment:
I'd prefer to see this `if` statement as part of `translate_key_to_cfg`, and let this code throw in the shortcut unconditionally.